### PR TITLE
QR-code reader refactoring [Qt6]

### PR DESCRIPTION
**Pull request by alexbruy**
Misc fixes for QR-code reader:
- Make brackets square
- Open QR-code scanner in full screen
- Init QR-code scanner only when QR icon is clicked

**How it works now**  
- Android - we use an external app to scan code (ZXing - Barcode Scanner) - will be removed once https://bugreports.qt.io/browse/QTBUG-109168 is released and we migrate to Qt 6.5
- iOS and other platforms - we use ZXing lib

- As a part of refactoring CodeReader is now instantiated _only_ when requested (when user clicks on the QR logo in form). It is opened fullscreen and destroys itself when the code is scanned.

> Note: there is one random crash, we need to confirm if it is related to Qt or not.

- [x] fix crash on iOS <-- not really, we still face https://bugreports.qt.io/browse/QTBUG-110131 (see https://github.com/MerginMaps/input/issues/2495)

UPDATE

The code for decoding has been simplified, no threading is used.
Camera is now lazy loaded ~> instantiated only when requested and **not destroyed afterwards**. It is achieved through QML singleton class `MMCamera.qml`

We are ready for review/testing

### CODE REVIEW

File: app/qml/CodeScannerOverlay.qml [hunk 1]
```diff
@@ -1,178 +1,98 @@
...
+
+      Repeater {
+        model: [0, 1, 3, 2]
+
+        delegate: Item {
+          property var length: 30  * __dp
+          property var borderWidth: 5 * __dp
+          property var animationDuration: 1000
+          property var primaryColor: InputStyle.fontColor
+          property var secondaryColor: InputStyle.fontColorBright
```
Comment (tomasMizera) [inline app/qml/CodeScannerOverlay.qml:36]: Let's avoid `var` types, they can not be compiled
 
```suggestion
          property real length: 30  * __dp
          property real borderWidth: 5 * __dp
          property int animationDuration: 1000
          property color primaryColor: InputStyle.fontColor
          property color secondaryColor: InputStyle.fontColorBright
```

File: app/attributes/attributeformmodel.cpp [hunk 2]
```diff
@@ -165,6 +165,11 @@ bool AttributeFormModel::setData( const QModelIndex &index, const QVariant &valu
 
     case AttributeValue:
     {
+      const FormItem * item = mController->formItem( uuid );
+      if ( mController->formValue( item->fieldIndex() ) == value )
+      {
+        return false;
+      }
```
Comment (tomasMizera) [inline app/attributes/attributeformmodel.cpp:172]: Just a note, we need to test this properly to see if it won't break forms

File: app/qml/CodeScanner.qml [hunk 3]
```diff
@@ -11,72 +11,74 @@ import QtQuick
...
 import QtMultimedia
-import lc 1.0
+import lc 1.0 as InputClass
 
 import "./components"
 
-Drawer {
-  id: codeReader
-  palette.dark: InputStyle.fontColor // changes busy indicator color
+Item {
```
Comment (tomasMizera) [inline app/qml/CodeScanner.qml:18]: Let's make the `Page` a top-level element. There should be no need to wrap it inside an `Item`

File: app/qml/editor/inputqrcodereader.qml [hunk 4]
```diff
@@ -72,5 +74,29 @@ AbstractEditor {
...
+  Component {
+    id: readerComponent
+
+    CodeScanner {
+      id: codeScannerPage
+
+      focus: true
+
+      onBackButtonClicked: {
+        root.parent.formView.pop()
```
Comment (tomasMizera) [inline app/qml/editor/inputqrcodereader.qml:93]: Can you please declare the stackview as a property? 
`property StackView formView`

**Review by tomasMizera — COMMENTED** [review]
Looks really good so far 🚀

File: app/imageutils.cpp [hunk 5]
```diff
@@ -99,37 +138,24 @@ bool ImageUtils::rescale( const QString &path, int quality )
...
-    return false;
-  }
+  return rescaledImage;
+}
 
-  QFileInfo fi( path );
-  QString newPath = QStringLiteral( "%1/%2_rescaled.%3" ).arg( fi.path(), fi.baseName(), fi.completeSuffix() );
+QImage ImageUtils::rescaledImage( const QVideoFrame &videoFrame )
+{
+  qDebug() << "ImageUtils::image";
```
Comment (alexbruy) [inline app/imageutils.cpp:146]: Probably not needed anymore.

File: app/qml/main.qml [hunk 6]
```diff
@@ -9,17 +9,14 @@
 
 import QtQuick
 import QtQuick.Controls
-
-// Required for iOS to get rid of "module "QtMultimedia" is not installed".
-// It looks like static QT plugins are not copied to the distribution
 import QtMultimedia
-import QtQml.Models
+//import QtQml.Models
```
Comment (alexbruy) [inline app/qml/main.qml:13]: Remove commented line?

File: app/codescanner.cpp [hunk 7]
```diff
@@ -1,159 +1,81 @@
...
   }
-  pauseProcessing();
-}
 
-void CodeScanner::processImage( QRDecoder *decoder, const QImage &image )
-{
-  decoder->processImage( image );
-  continueProcessing();
-}
+  QImage image = ImageUtils::rescaledImage( frame );
```
Comment (alexbruy) [inline app/codescanner.cpp:33]: Can we have a check for image validity here? In some cases `rescaledImage` returns empty image and there is no sense in trying to process it.

**Review by alexbruy — COMMENTED** [review]
Looks good to me

**Comment by alexbruy** [comment]
Looks good to me.

**Pull request merged**